pkg/storageincentives/agent.go

```diff
@@ -472,3 +493,23 @@ func (a *Agent) wrapCommit(storageRadius uint8, sample []byte, key []byte) ([]by
 func (a *Agent) Status() (*Status, error) {
 	return a.state.Status()
 }
+
+func (a *Agent) HasEnoughFundsToPlay(ctx context.Context) (bool, error) {
+	balance, err := a.backend.BalanceAt(ctx, a.state.ethAddress, nil)
+	if err != nil {
+		return false, err
+	}
+
+	price, err := a.backend.SuggestGasPrice(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	avgGasPrice := big.NewInt(avgTxGas)
+	avgGasPrice = avgGasPrice.Mul(avgGasPrice, price)
```

Comment:
So I am not clear on this. Why do we multiply `avgGasPrice * price`? I was thinking this should just be the `price * minTxCountToCover`

Reply:
the code here does need some rewording, but the calculation is correct.
the miminum balance  = currentPrice * avgGas * nTransactionToCover

```
avgTxFee := new(big.Int).Mul(big.NewInt(avgTxGas), price)
minBalance :=  avgTxFee.Mul(avgTxFee, big.NewInt(minTxCountToCover))
```